src/main/java/com/temporal/samples/helloworld/Client.java

```diff
@@ -57,29 +66,44 @@ public Client(Environment env) throws IOException {
 
         // Load your client certificate, which should look like:
         InputStream clientCert = new FileInputStream(env.getProperty("temporal.tls.client.certPath"));
+        
         // PKCS8 client key, which should look like:
         InputStream clientKey = new FileInputStream(env.getProperty("temporal.tls.client.keyPath"));
-        // certification Authority signing certificate
+        
+        // Certification Authority signing certificate
         InputStream caCert = new FileInputStream(env.getProperty("temporal.tls.ca.certPath"));
 
-        // Create a TLS Channel Credential Builder : https://community.temporal.io/t/how-to-disable-host-name-verification/2808/8
-        var tlsBuilder = TlsChannelCredentials.newBuilder();
-        tlsBuilder.keyManager(clientCert, clientKey);
-        tlsBuilder.trustManager(caCert);
-        var channel = Grpc.newChannelBuilder(temporalServerUrl, tlsBuilder.build())
-                .overrideAuthority(temporalServerCertAuthorityName)
-                .build();
+        // Create Trust Manager
+        KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        trustStore.load(null, null);
+        X509Certificate certificate = (X509Certificate)CertificateFactory
+            .getInstance("X509")
+            .generateCertificate(caCert);
+        trustStore.setCertificateEntry("ca", certificate);
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory

```

Comment:
The current impelementation still more clearer and compacted over the new one, I will approved it for now and later we can have a deeper discussion